src/axes/axis.ts

```diff
@@ -150,9 +168,216 @@ export class Axis<D> extends Component {
+      case "bottom":
+        secondaryPosition = 0;
+        break;
+      case "top":
+        secondaryPosition = this.height();
+        break;
+      case "left":
+        secondaryPosition = this.width();
+        break;
+      case "right":
+        secondaryPosition = 0;
+        break;
+    }
+
+    bindElements(this._annotationContainer.select(".annotation-line-container"), "line", "annotation-line")
+      .attr(this._isHorizontal() ? "x1" : "y1", positionF)
+      .attr(this._isHorizontal() ? "x2" : "y2", positionF)
+      .attr(this._isHorizontal() ? "y1" : "x1", secondaryPosition)
+      .attr(this._isHorizontal() ? "y2" : "x2", offsetF)
+      .attr("visibility", visibilityF);
+
+    bindElements(this._annotationContainer.select(".annotation-circle-container"), "circle", "annotation-circle")
+      .attr(this._isHorizontal() ? "cx" : "cy", positionF)
+      .attr(this._isHorizontal() ? "cy" : "cx", secondaryPosition)
+      .attr("r", 3)
+      .attr("visibility", visibilityF);
+
+    let rectangleOffsetF = (d: D) => {
+      switch (this.orientation()) {
+        case "bottom":
+        case "right":
+          return offsetF(d);
+        case "top":
+        case "left":
+          return offsetF(d) - measurements.get(d).height;
+      }
+    };
+    bindElements(this._annotationContainer.select(".annotation-rect-container"), "rect", "annotation-rect")
+      .attr(this._isHorizontal() ? "x" : "y", positionF)
+      .attr(this._isHorizontal() ? "y" : "x", rectangleOffsetF)
+      .attr(this._isHorizontal() ? "width" : "height", (d) => measurements.get(d).width)
+      .attr(this._isHorizontal() ? "height" : "width", (d) => measurements.get(d).height)
+      .attr("visibility", visibilityF);
+
+    let annotationWriter = this._annotationWriter;
+    let annotationFormatter = this.annotationFormatter();
+    let isHorizontal = this._isHorizontal();
+    bindElements(this._annotationContainer.select(".annotation-label-container"), "g", "annotation-label")
+      .attr("transform", (d) => {
+        let xTranslate = this._isHorizontal() ? positionF(d) : rectangleOffsetF(d);
+        let yTranslate = this._isHorizontal() ? rectangleOffsetF(d) : positionF(d);
+        return "translate(" + xTranslate + "," + yTranslate + ")";
+      })
+      .attr("visibility", visibilityF)
+      .each(function (annotationLabel) {
+        let writeOptions = {
+          selection: d3.select(this),
+          xAlign: "center",
+          yAlign: "center",
+          textRotation: isHorizontal ? 0 : 90
+        };
+        annotationWriter.write(annotationFormatter(annotationLabel),
+                                 isHorizontal ? measurements.get(annotationLabel).width : measurements.get(annotationLabel).height,
+                                 isHorizontal ? measurements.get(annotationLabel).height : measurements.get(annotationLabel).width,
+                                 writeOptions);
+      });
+  }
+
+  protected _axisHeightWithoutMargin() {
```

Comment:
Maybe "`_axisSizeWithoutMargin()`" or some other name that doesn't suggest an orientation?